### DISCUSSION TITLE
[BUG] cannot push to protected branch with user supposed to be allowed to

### DISCUSSION THREAD

**Discussion by Roms1383** [community/community#48232] [Actions]
### Select Topic Area

Bug

### Body

Hi ! It seems branch protection rules cannot be bypassed when they are supposed to, or did I miss something ?

## context

- organization repo: [cyb3rpsych0s1s/4ddicted](https://github.com/cyb3rpsych0s1s/4ddicted) (that I own)
- branch protection rules:
   as you can see I require a PR with approval but also set my personal bot [Bender](https://github.com/kroms-bot) to be allowed to both bypass protection rule **and** to force push. I also set his Personal Access Token both in the organization's Secrets and as an env var in the [failing workflow job](https://github.com/cyb3rpsych0s1s/4ddicted/blob/7a3a367ba35d72fa7b7503763d1b98e03c4a2195/.github/workflows/release.yml#L75-L100)
   
<img width="768" alt="Screenshot 2566-02-22 at 10 52 50" src="https://user-images.githubusercontent.com/21016014/220518906-6cb5dd01-cac3-4316-8ce5-f017fb4dd453.png">
<img width="755" alt="Screenshot 2566-02-22 at 10 53 15" src="https://user-images.githubusercontent.com/21016014/220518928-3b7ed7a0-cff5-414c-8b9a-abd571f259cc.png">
<img width="751" alt="Screenshot 2566-02-22 at 10 57 06" src="https://user-images.githubusercontent.com/21016014/220518933-401868ab-d8b8-4798-b6ad-dd3a6ab514a4.png">

but as you can see from my [latest Actions runs](https://github.com/cyb3rpsych0s1s/4ddicted/actions), it doesn't get allowed.

I have done a search for a similar discussion and tried the solutions suggested there (e.g. `[skip ci]`) but it didn't solve.

Is there any way to fix this ?

**Comment by Roms1383** [comment]
I'm gonna disable the `changelog` action for now as I need to keep releasing, so as a reminder the related Actions are numbered: [101](https://github.com/cyb3rpsych0s1s/4ddicted/actions/runs/4238598669), [103](https://github.com/cyb3rpsych0s1s/4ddicted/actions/runs/4238672170), [105](https://github.com/cyb3rpsych0s1s/4ddicted/actions/runs/4238940529), [107](https://github.com/cyb3rpsych0s1s/4ddicted/actions/runs/4239106449) and latest [109](https://github.com/cyb3rpsych0s1s/4ddicted/actions/runs/4239141311).

**Comment by yesh4gvm** [comment]
Did you find any solution for same ? @Roms1383 . I am facing same issue

**Reply by Roms1383** [reply]
Looking at my commits I think I just removed it. Haven't retried ever since @yesh4gvm.

**Reply by yesh4gvm** [reply]
ok @Roms1383

**Comment by anthms** [comment]
Hi @Roms1383 👋 - the authentication used for `git` operations run directly through the terminal is configured by the `actions/checkout` step:
https://github.com/actions/checkout/blob/main/action.yml#L12-L24

A `token` parameter can be passed, otherwise `GITHUB_TOKEN` is used by default - which will authenticate further `git` commands in the job as the `github-actions` bot.

If you set the relevant PAT to the `token` parameter for `actions/checkout` - the `git` operation should pass with bypassing the protection rule requirement.

```
steps:
      - uses: actions/checkout@v3
        with:
          token: ${{ secrets.KROMS_BOT_TOKEN }}
          fetch-depth: 0
      ...
      - name: Regenerate CHANGELOG.md
        run: |
          git config user.name "kroms-bot"
          git config user.email ""
          ...
```

**Reply by Roms1383** [reply]
Oh ok I'll have to give a try again in the coming weeks, thanks @anthms.

**Comment by joshjohanning** [comment]
Do you have _required status checks_ enabled?  If you enable _required status checks_, then even users/apps that you add to your list won't be able to bypass directly pushing without creating a pull request. 

What you can do is have that user create a pull request and then they can use the fact that they are added to the list to override the merge. 

![image](https://github.com/community/community/assets/19912012/a2632f0c-43e2-49b2-8c87-1e72d2d4aec5)

![image](https://github.com/community/community/assets/19912012/94ce00f8-f779-4a4f-bf05-351fd3e2a77f)

**Comment by bl4ko** [comment]
For me a way to make this work was to use a **Ruleset** feature instead of **classic branch protection rule**.

You can make ruleset with the same options as classic branch protection rule, but in the bypass list, you must select **Write role**. Now all collaborators will be able to force push (your bot in this case), even if they don't meet the requirements for the push.
